docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,488 @@
+---
+id: higher-order-components
+title: Higher-Order Components
+permalink: docs/higher-order-components.html
+---
+
+A higher-order component (HOC) is an advanced technique in React for reusing component logic. HOCs are not part of the React API, per se. They are a pattern that emerges from React's compositional nature.
+
+Concretely, **a higher-order component is a function that takes a component and returns a new component.** An HOC accepts a component as one of its arguments, and returns an enhanced version of that component:
+
+```js
+const EnhancedComponent = higherOrderComponent(WrappedComponent);
+```
+
+Whereas a component transforms props into UI, a higher-order component transforms one component into another component.
```

Comment:
Getting nitpicky but I think this "component into a component" is pretty much repeated in the last two sentences as well. Can we just merge them into a single sentence?